packages/react-ai/src/hooks/useAIGeneration.tsx

```diff
@@ -74,10 +74,17 @@ export function createUseAIGeneration<
       undefined
     );
 
+    let { hasError, message } = result;
+
     const { data, errors } =
       (result?.data as SingularReturnValue<Schema[Key]['returnType']>) ?? {};
 
-    return [{ ...result, data, graphqlErrors: errors }, handler];
+    if (errors) {
+      hasError = true;
+      message = errors.map((error) => error.message).join(' ');
+    }
+
+    return [{ ...result, data, hasError, message }, handler];
```

Comment:
Can we ever get into a case where useDataState's result has an error `message` and we also have graphqlErrors? In that case graphqlErrors would override that `useDataState` message?

Though I think that would only happen in the case where our previous call had graphqlErrors, then our next generation resulted in an error

Reply:
I think we should be fine because the amplify data client (using appsync) will always return errors in the same way. I don't think there is another way to get an error from the data client.

Reply:
Thinking `useDataState` is the wrong approach here for a handful of reasons:
- `output.hasError` is `true`  when the API call itself fails, which is why it's `false` handled in this use case
- `output.message` exposes the error message of the failed API call
- afaict `handleGenerate` is setting a subscription in a fire and forget fashion rather than using  continuous calls to the `useDataState` handler to provide stateful `data`